packages/squiggle-lang/src/library/registry/frTypes.ts

```diff
@@ -133,6 +136,46 @@ export const frLambdaTyped = (
   };
 };
 
+//This will box if not boxed. We could do a form that doesn't do this, but it would be messier.
+//I could see it as cleaner to use a new Class or interface like {type: "Boxed", value: T, args: BoxedArgs} | {type: "Unboxed", value: T}, but doesn't seem worth it yet.
+export const frForceBoxed = <T>(
+  itemType: FRType<T>
+): FRType<{ value: T; args: BoxedArgs }> => {
+  return {
+    unpack: (v) => {
+      if (v.type !== "Boxed") {
+        const unpackedItem = itemType.unpack(v);
+        if (unpackedItem) {
+          return { value: unpackedItem, args: new BoxedArgs({}) };
+        } else {
+          return undefined;
+        }
+      } else {
+        const unpackedItem = itemType.unpack(v.value.value);
+        const boxedArgs: BoxedArgs = v.value.args;
+        return (
+          (unpackedItem && { value: unpackedItem, args: boxedArgs }) ||
+          undefined
+        );
+      }
+    },
+    pack: ({ value, args }) => vBoxed(new Boxed(itemType.pack(value), args)),
+    getName: itemType.getName,
+    keepBoxes: true,
+  };
+};
+
+//This works just like any, but it will preserve boxes, if items are boxed. Useful for "inspect" or cases where you don't need to unpack the underlying value.
+export function frKeepBoxes<T1>(t: FRType<T1>) {
```

Comment:
Why "keep boxes" and not "keep boxed" (or maybe `frPreserveBoxed`), if there's just a single maybe-box?